### PR TITLE
DPDK: fix broken OVS version check

### PR DESCRIPTION
**Pull request by mcgov**
Previously the test assumed an explicit version would be passed in, and would just fail if the DPDK and OVS versions didn't match up.

This commit switches to checking DPDK version first and then checking out the compatible OVS version to run the test.

### CODE REVIEW

File: microsoft/testsuites/dpdk/dpdkovs.py [hunk 1]
```diff
@@ -111,42 +111,49 @@ def _check_ovs_dpdk_compatibility(self, dpdk_tool: DpdkTestpmd) -> None:
...
-        # check if version too high (pairing is unknown)
-        elif self.ovs_version > "2.18.0":
-            raise NotImplementedError(
-                (
-                    f"Not implemented OVS version {self.ovs_version}. "
-                    "Please update the version requirement table in "
-                    "dpdkovs.py:check_ovs_dpdk_compatibilty with new data"
-                    "from https://docs.openvswitch.org/en/latest/faq/releases/"
+        else:
+            if dpdk_version < "19.11.0":
```
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdkovs.py]: Why low dpdk version skipped, but low ovs version failed?

File: microsoft/testsuites/dpdk/dpdkovs.py [hunk 2]
```diff
@@ -111,42 +111,49 @@ def _check_ovs_dpdk_compatibility(self, dpdk_tool: DpdkTestpmd) -> None:
...
+                raise SkippedException(
+                    f"Dpdk version {dpdk_version} is not supported by this test."
+                )
+
+            if int(dpdk_version.major) not in dpdk_to_ovs_minimum_version.keys():
+                # we've already checked out latest OVS
+                # DPDK version is above 22, warn and proceed.
+                self.node.log.warn(
```
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdkovs.py]: info level should be enough, actually, nobody is looking for warning message.

File: microsoft/testsuites/dpdk/dpdkovs.py [hunk 3]
```diff
@@ -111,42 +111,49 @@ def _check_ovs_dpdk_compatibility(self, dpdk_tool: DpdkTestpmd) -> None:
...
+                    "DPDK version is above the maximum in the version match"
+                    " table. Using latest OVS."
                 )
+                return
```
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdkovs.py]: Does it mean not to run following ovs enlistment code? So what happens, if return here?

File: microsoft/testsuites/dpdk/dpdkovs.py (hunk 1 above)
Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkovs.py]: Hmm good point! I'll skip.

File: microsoft/testsuites/dpdk/dpdkovs.py (hunk 3 above)
Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkovs.py]: OVS is already checking out latest at the top of the test, so if the DPDK version is also above the levels in our table (22 and beyond) it should be fine to just run with what's already checked out. I'll comment this a little better.
I don't really like that this table has to be updated constantly, but if it starts failing we'll see the note in the log and know that the test might need an update to the table.

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkovs.py]: The alternative is pulling the rst table from the docs and parsing it though, so this seems cleaner than that.

File: microsoft/testsuites/dpdk/dpdkovs.py (hunk 1 above)
Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkovs.py]: I forgot this wasn't even needed with this change. We check out latest, check the installed dpdk version, adjust OVS version lower if needed. OVS won't ever be too low of a version so we can just ditch that entirely.

**Review by squirrelsc — APPROVED** [review]

File: microsoft/testsuites/dpdk/dpdkovs.py (hunk 3 above)
Comment (squirrelsc) [reply microsoft/testsuites/dpdk/dpdkovs.py]: Fantastic version compatibility handling!

**Pull request merged**